pysolarmanv5/pysolarmanv5.py

```diff
@@ -102,14 +103,15 @@ def __init__(self, address, serial, **kwargs):
         self.sock: socket.socket = kwargs.get("socket", self._create_socket())
         if self.sock is None:
             raise NoSocketAvailableError("No socket available")
-        self._poll = select.poll()
-        self._sock_fd = self.sock.fileno()
-        self._auto_reconnect = False if kwargs.get('socket') else kwargs.get('auto_reconnect', False)
-        self._data_queue = Queue(maxsize=1)
-        self._data_wanted = Event()
-        self._reader_exit = Event()
-        self._reader_thr = Thread(target=self._data_receiver, daemon=True)
-        self._reader_thr.start()
+        if isinstance(self.sock, socket.socket):
+            self._poll = select.poll()
+            self._sock_fd = self.sock.fileno()
+            self._auto_reconnect = False if kwargs.get('socket') else kwargs.get('auto_reconnect', False)
+            self._data_queue = Queue(maxsize=1)
+            self._data_wanted = Event()
+            self._reader_exit = Event()
+            self._reader_thr = Thread(target=self._data_receiver, daemon=True)
+            self._reader_thr.start()
```

Comment:
Rather than passing `sock` as a `str` PySolarmanV5Async to avoid creating the duplicate socket, how about we move the socket stuff from the PySolarmanV5 constructor into a separate _socket_init_ function, and call that new function within the the constructor. On the PySolarmanV5Async side, override the _socket_init_ function so that is a noop.